src/modules/Dashboard/features/Managers/Popups/AddManagerPopup/AddManagerPopup.tsx

```
@@ -0,0 +1,227 @@
+  // Workspace Name field is only available if the workspace has no managers yet
+  const isWorkspaceNameVisible = !!workspaceInfo && !workspaceInfo.hasManagers;
+
+  const { ownerId } = workspaces.useData() || {};
+  const respondentsData = users.useAllRespondentsData();
+  const participants = (respondentsData?.result ?? []).map(({ details }) => {
+    const { respondentSecretId, respondentNickname, subjectId } = details[0];
+
+    return {
+      subjectId,
+      secretId: respondentSecretId,
+      nickname: respondentNickname,
+      // TODO: Populate tag
+      // https://mindlogger.atlassian.net/browse/M2-5861
+      tag: null,
+    };
+  });
+
+  const defaults = {
+    ...defaultValues,
+    language: i18n.language as Languages,
+    workspaceName: workspaceInfo?.name,
+  };
+
+  const {
+    control,
+    formState: { isValid, isDirty },
+    handleSubmit,
+    register,
+    reset,
+    setError,
+    setValue,
+    unregister,
+  } = useForm<AddManagerFormValues>({
+    resolver: yupResolver(AddManagerPopupSchema(isWorkspaceNameVisible)),
+    defaultValues: defaults,
+    mode: 'all',
+    delayError: VALIDATION_DEBOUNCE_VALUE,
+  });
+
+  const role = useWatch({ control, name: Fields.role });
+
+  const dispatch = useAppDispatch();
+  const [hasCommonError, setHasCommonError] = useState(false);
+
+  const handleClose = (shouldRefetch = false) => {
+    resetForm();
+    setHasCommonError(false);
+    onClose?.(shouldRefetch);
+  };
+
+  const resetForm = () => reset(defaults);
+
+  const {
+    error,
+    execute: createInvitation,
+    isLoading,
+  } = useAsync(postAppletInvitationApi, async (result) => {
+    ownerId && executeGetWorkspaceInfoApi({ ownerId });
+    const { firstName, lastName, title } = result.data?.result ?? {};
+    dispatch(
+      banners.actions.addBanner({
+        key: 'AddParticipantSuccessBanner',
+        bannerProps: {
+          id: `${firstName} ${lastName}${title ? `, ${title}` : ''}`,
+        },
+      }),
+    );
```

I've read the description and comments on [M2-6159](https://mindlogger.atlassian.net/browse/M2-6159) and agree that it would be ideal for these fields to be coming back in the response. However, I don't think we should leave the UX like this in the interim. At this point we have a successful response as well as the values that were submitted. I think it's reasonable to assume the values were accepted and display them to the user in the banner. If there is an API validation error, it should return an error response

---

Last time I tried to work around this issue by injecting the form values into the banner, they still didn't come through, and I assumed it was because the popup was getting destroyed too soon (in the context of `AddParticipantPopup`), but it doesn't look like that should be happening here. So you're right, I _should_ be able to get the values straight from the form without issue – I'll give it a try.

---

Workaround no longer required, fixed in https://github.com/ChildMindInstitute/mindlogger-backend-refactor/pull/1273!